### ISSUE TITLE
decode() misparses end-of-file “relaxed” comment

### ISSUE THREAD

**Issue by FGasper** [rurban/Cpanel-JSON-XS#174]
```
> perl -MCpanel::JSON::XS -e'print $Cpanel::JSON::XS::VERSION . $/; print Cpanel::JSON::XS->new()->relaxed(1)->decode(q<["ha"] # ohyeah>)'
4.24
garbage after JSON object, at character offset 16 (before "") at -e line 1.
```
^^ The above should parse, right?

JSON::XS 4.x has the same problem, though 3.x parsed it.

**Comment by rurban** [comment]
I'm not sure if it should parse, but it's definitely a regression

**Comment by rurban** [comment]
Fixed with 4.25
